src/ngraph/ngraph_utils.cc

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+* Copyright 2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "ngraph_utils.h"
+
+#include <algorithm>
+#include <stdexcept>
+
+using namespace std;
+
+namespace ngraph_bridge {
+
+std::ostream& operator<<(std::ostream& os, const ngraph::Shape& s) {
+  os << "[";
+  for (size_t d : s) {
+    os << " " << d;
+  }
+  os << "]";
+  return os;
+}
```

Comment:
I've done this a lot while debugging, the utilty is nice. Can we templatize it to support nnvm::Shape as well?

Reply:
I've added an overload to handle `nnvm::TShape`.  I'm hesitant to templatize this in a way that picks up both `nnvm::TShape` and `ngraph::Shape`, because I fear such a template would bind to uses that we never intended.